src/main/java/de/sakpaas/backend/service/LocationApiSearchDas.java

```diff
@@ -11,29 +12,48 @@
 @Component
 public class LocationApiSearchDas {
 
+
   /**
-   * Gets all Locations in a Country (currently only Germany).
+   * Gets all Locations of specific types in a specific Country.
    *
-   * @param countryCode ISO3166-1:alpha2 Code
+   * @param osmImportConfiguration Object which contains all information about the data to load
    * @return list of supermarkets in Country
    */
   public List<OsmResultLocationListDto.OsmResultLocationDto> getLocationsForCountry(
-      String countryCode) {
-    final String url =
-        "https://overpass-api.de/api/interpreter?data=[out:json][timeout:2500];"
-            + "area[\"ISO3166-1:alpha2\"=" + countryCode + "]->.searchArea;("
-            + "node[shop=supermarket](area.searchArea);way[shop=supermarket](area.searchArea);"
-            + "node[shop=chemist](area.searchArea);way[shop=chemist](area.searchArea);"
-            + "node[shop=beverages](area.searchArea);way[shop=beverages](area.searchArea);"
-            + ");out center;";
+
+      OsmImportConfiguration osmImportConfiguration) {
+
+    // If no location types specified, there is nothing to load
+    if (osmImportConfiguration.getShoptypes().isEmpty()) {
+      return emptyList();
+    }
+
+    // build request string
+    StringBuilder url =
+        new StringBuilder("https://overpass-api.de/api/interpreter?data=[out:json][timeout:2500];")
+            .append("area[\"ISO3166-1:alpha2\"=").append(osmImportConfiguration.getCountry())
+            .append("]->.searchArea;(");
+
+    // Add shoptypes from configuration
+    for (String shoptype : osmImportConfiguration.getShoptypes()) {
+      url.append("node[shop=").append(shoptype).append("](area.searchArea);way[shop=")
+          .append(shoptype).append("](area.searchArea);");
+    }
+
+    url.append(");out center;");
+
+    // make request
+
+    
     RestTemplate restTemplate = new RestTemplate();
     ResponseEntity<OsmResultLocationListDto> response =
-        restTemplate.getForEntity(url, OsmResultLocationListDto.class);
+        restTemplate.getForEntity(url.toString(), OsmResultLocationListDto.class);
```

Comment:
warum tust du das? xD

Reply:
Muss man, weil url ein Stringbuilder ist ^^

Reply:
Habs jz in ne extra Funktion ausgelagert